test/test_dagnav.py

```diff
@@ -157,6 +151,99 @@ def test_group_create(request):
     assert model.groups_by_name['mat:plastic'] == new_group2
     assert len(model.groups) == orig_num_groups + 2
 
+def test_group_name_runtime_error():
+    """Test group name returns None when tag is missing."""
+    model = pydagmc.Model()
+    group = model.create_group(group_id=1)
+    try:
+        model.mb.tag_delete(model.name_tag, (group.handle,))
+    except RuntimeError:
+         pass
+    assert group.name is None
+
+def test_group_name_setter_duplicate():
+    """Test assigning an already used group name."""
+    model = pydagmc.Model()
+    group1 = model.create_group(name='group_a', group_id=1)
+    group2 = model.create_group(name='group_b', group_id=2)
+    with pytest.raises(ValueError, match="Group group_a already used"):
+        group2.name = 'group_a'
+
+def test_group_repr_empty():
+    """Test Group.__repr__ with no volumes/surfaces."""
+    model = pydagmc.Model()
+    group_empty = model.create_group(name='empty_group', group_id=1)
+    repr_empty = repr(group_empty)
+    assert 'Name: empty_group' in repr_empty
+    assert 'Volume IDs:' not in repr_empty
+    assert 'Surface IDs:' not in repr_empty
+
+    group_surf_only = model.create_group(name='surf_only', group_id=2)
+    surf = model.create_surface(global_id=10)
+    group_surf_only.add_set(surf)
+    repr_surf_only = repr(group_surf_only)
+    assert 'Name: surf_only' in repr_surf_only
+    assert 'Volume IDs:' not in repr_surf_only
+    assert 'Surface IDs:' in repr_surf_only
+    assert '10' in repr_surf_only
+
+    group_vol_only = model.create_group(name='vol_only', group_id=3)
+    vol = model.create_volume(global_id=20)
+    group_vol_only.add_set(vol)
+    repr_vol_only = repr(group_vol_only)
+    assert 'Name: vol_only' in repr_vol_only
+    assert 'Volume IDs:' in repr_vol_only
+    assert '20' in repr_vol_only
+    assert 'Surface IDs:' not in repr_vol_only
+
```

Comment:
PEP8: Two blank lines between file-level declarations. Let's update to this throughout.

```suggestion


```